### PR TITLE
Verify we don't break the ABI

### PR DESCRIPTION
**Pull request by hughsie**
This adds a script which can check for ABI breaks between two Git
revisions:

    $ ./contrib/ci/check-abi.sh

The CI is set up to run it automatically between the tip of the branch
being tested and the last release tag.

Based on the work by Mathieu Bridon <bochecha@daitauha.fr>, many thanks.

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: contrib/ci/check-abi [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                              env={'DESTDIR': dest_dir})
+
+    return dest_dir
+
+
+def compare(old_tree, new_tree):
+    print(format_title(f'# Comparing the two ABIs'), end='\n\n', flush=True)
+
+    old_headers = os.path.join(old_tree, 'usr', 'include')
+    old_lib = os.path.join(old_tree, 'usr', 'lib', 'libxmlb.so')
```
Comment (superm1) [inline contrib/ci/check-abi]: Isn't this checking the wrong library?  I figure you want to check libfwupd.so

File: contrib/ci/check-abi [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        print("Let's not waste time comparing something to itself")
+        sys.exit(0)
+
+    old_tree = build_install(args.old)
+    new_tree = build_install(args.new)
+
+    try:
+        compare(old_tree, new_tree)
+
+    except Exception:
```
Comment (superm1) [inline contrib/ci/check-abi]: You should specifically be catching `CalledProcessError` shouldn't you?

**Comment by superm1** [comment]
Considering the challenges you're hitting, what about just appending this to the tail of one of the other builds?  Like maybe s390x or so?

**Comment by hughsie** [comment]
I suck at travis stuff. If you know what you're doing more than me all help very welcome.

**Comment by superm1** [comment]
@hughsie I've got it working now in circle CI if that's OK with you.

**Comment by hughsie** [comment]
Much appreciated! LGTM.

**Pull request merged**